xbmc/addons/AddonDatabase.cpp

```diff
@@ -845,7 +847,7 @@ bool CAddonDatabase::Search(const std::string& search, VECADDONS& addons)
     while (!m_pDS->eof())
     {
       AddonPtr addon;
-      GetAddon(m_pDS->fv(0).get_asString(),addon);
+      GetAddon(m_pDS->fv(0).get_asInt(), addon);
```

Comment:
Yes this change makes sense as with string addon id more hard to identify related one and always take the first.